src/main/java/org/spongepowered/api/command/args/GenericArguments.java

```diff
@@ -1585,4 +1604,277 @@ public Text getUsage(CommandSource src) {
         }
     }
 
+    /**
+     * Expect an argument to represent a {@link URL}.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement url(Text key) {
+        return new URIElement(key, false);
+    }
+
+    /**
+     * Expect an argument to represent a {@link URI}.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement uri(Text key) {
+        return new URIElement(key, true);
+    }
+
+    private static class URIElement extends KeyElement {
+
+        private final boolean returnURI;
+
+        protected URIElement(Text key, boolean returnURI) {
+            super(key);
+            this.returnURI = returnURI;
+        }
+
+        @Nullable
+        @Override
+        protected Object parseValue(CommandSource source, CommandArgs args) throws ArgumentParseException {
+            String str = args.next();
+            URL url;
+            try {
+                url = new URL(str);
+            } catch (MalformedURLException ex) {
+                throw new ArgumentParseException(Text.of("Invalid URL!"), ex, str, 0);
+            }
+            URI uri;
+            try {
+                uri = url.toURI();
+            } catch (URISyntaxException ex) {
+                throw new ArgumentParseException(Text.of("Invalid URL!"), ex, str, 0);
+            }
+            if (this.returnURI) {
+                return uri;
+            } else {
+                return url;
+            }
+        }
+    }
+
+    /**
+     * Expect an argument to return an IP address, in the form of an
+     * {@link InetAddress}.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement ip(Text key) {
+        return new IpElement(key, false);
+    }
+
+    /**
+     * Expect an argument to return an IP address, in the form of an
+     * {@link InetAddress}, or if nothing matches and the source is a
+     * {@link RemoteSource}, return the source's address.
+     *
+     * @param key The key to store under
+     * @return the argument
+     */
+    public static CommandElement ipOrSource(Text key) {
```

Comment:
That's actually a very good idea, I'll look into that. That's one that would get used plenty (and by Reveries, no less :P)